truffle/com.oracle.truffle.api.vm/src/com/oracle/truffle/api/vm/PolyglotEngine.java

```diff
@@ -80,18 +80,26 @@
  * {@link PolyglotEngine} is in inter-operability between all Truffle languages. There is 1:N
  * mapping between {@link PolyglotEngine} and {@link TruffleLanguage Truffle language
  * implementations}.
+ *
+ * <h2>Usage</h2>
+ *
  * <p>
- * Use {@link #buildNew()} to create new isolated portal ready for execution of various languages.
- * All the languages in a single portal see each other exported global symbols and can cooperate.
- * Use {@link #buildNew()} multiple times to create different, isolated portal environment
+ * Use {@link #buildNew()} to create a new isolated portal ready for execution of various languages.
+ * All the languages in a single portal see each others exported global symbols and can cooperate.
+ * Use {@link #buildNew()} multiple times to create different, isolated environments that are
  * completely separated from each other.
  * <p>
  * Once instantiated use {@link #eval(com.oracle.truffle.api.source.Source)} with a reference to a
  * file or URL or directly pass code snippet into the virtual machine via
  * {@link #eval(com.oracle.truffle.api.source.Source)}. Support for individual languages is
  * initialized on demand - e.g. once a file of certain MIME type is about to be processed, its
  * appropriate engine (if found), is initialized. Once an engine gets initialized, it remains so,
- * until the virtual machine isn't garbage collected.
+ * until the virtual machine is garbage collected.
+ * <p>
+ * A {@link com.oracle.truffle.api.TruffleLanguage} should be designed to be initializable and
+ * configurable in a generic way. Language-specific execution front-ends, for instance to emulate
+ * existing language implementations, should be realized externally to maintain a consistent and
+ * language-agnostic way of initializing {@link PolyglotEngine}.
```

Comment:
It would be better if the documentation of `PolyglotEngine` was more focused on users of the languages. For example:

```
If the way you use a language requires special setup,
you should obtain necessary parameters from somewhere (for example 
when parsing a command line) and communicate them to
the {@link PolyglotEngine} - either via {@link PolyglotEngine.Builder#config} 
as exposed by the language implementation or by evaluating 
appropriate "prelude" scripts via {@link PolyglotEngine.Language#eval}.
Possibly you can pre-register various {@link PolyglotEngine.Builder#globalSymbol global objects}
and make them available to the system of {@link PolyglotEngine}.
```